### PR TITLE
PT-346: Performance optimization: arrayless rewards

### PR DESCRIPTION
**Pull request by akak1977**
Can boost promotion/rewards calculation by 3-5% (15% together with PR)
https://github.com/VirtoCommerce/vc-module-core/pull/179

https://virtocommerce.atlassian.net/browse/PT-283

### CODE REVIEW

File: src/VirtoCommerce.MarketingModule.Core/Model/Promotions/PromotionConditionAndRewardTree.cs [hunk 1]
```diff
@@ -11,12 +11,13 @@ public PromotionConditionAndRewardTree()
             All = true;
         }
 
-        public virtual PromotionReward[] GetRewards()
+        public virtual IEnumerable<PromotionReward> GetRewards()
```
[the remark this answers is not in the document]
Comment (Vectorfield4) [reply src/VirtoCommerce.MarketingModule.Core/Model/Promotions/PromotionConditionAndRewardTree.cs:14]: @tatarincev needs your review.

[the remark this answers is not in the document]
Comment (akak1977) [reply src/VirtoCommerce.MarketingModule.Core/Model/Promotions/PromotionConditionAndRewardTree.cs:14]: 100% breaking change. But what/who is affected?

**Review by tatarincev — COMMENTED** [review]
I think the resulting performance improvements don't worth these breaking changes. I've checked the performance results and they are on edge of statistic inaccuracies.  

But I  don't mind to try  parallelize the process of promo evaluation 
https://github.com/VirtoCommerce/vc-module-marketing/blob/dev/src/VirtoCommerce.MarketingModule.Data/Services/EvaluationPolicies/BestRewardPromotionPolicy.cs#L37

lest say if we split this sequential loop into two or three threads, this might be very interesting from the performance perspective.

**Comment by Vectorfield4** [comment]
Closed cause breaking changes.

**Pull request closed without merging**